modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/filter/FeatureChainedAttributeVisitor.java

```diff
@@ -265,6 +269,13 @@ private boolean isXlinkHref(StepList steps) {
         return steps.get(steps.size() - 1).getName().equals(XLINK.HREF);
     }
 
+    static boolean isFid(StepList steps) {
+        if (steps.isEmpty()) {
+            return false;
+        }
+        return steps.get(steps.size() - 1).toString().matches("@(\\w+:)?id");
```

Comment:
This assumption will be right most of the time, but not always. This regex assumes that `@id` and anything like `@ns:id` are an FID. I am sure that some schema will be written that has an `@ns:id` that is not an FID. And the regex does not match `fid`; does that matter?. Ugly, but I cannot think of a better solution.